x/programs/rust/wasmlanche-sdk/src/program.rs

```diff
@@ -6,11 +6,35 @@ use crate::{
     Gas,
 };
 use borsh::{BorshDeserialize, BorshSerialize};
-use std::{cell::RefCell, collections::HashMap};
+use std::{cell::RefCell, collections::HashMap, io::Read};
 use thiserror::Error;
 
+/// Defer deserialization from bytes
+#[cfg_attr(feature = "debug", derive(Debug))]
+pub struct DeferDeserialize(Vec<u8>);
+
+impl DeferDeserialize {
+    /// # Errors
+    /// Returns a [`std::io::Error`] if there was an issue deserializing the value
+    #[allow(clippy::must_use_candidate)]
```

Comment:
Do you need to do this? `Result` is already `must_use`.

Reply:
Good catch, looks like I don't because this function used to be infaillible.